### PR TITLE
IRIS-6474 | make highlighting great again

### PR DESCRIPTION
**Pull request by xkxd**
https://wikia-inc.atlassian.net/browse/IRIS-6474

@Wikia/iris

### CODE REVIEW

File: src/modules/Hyperlinking.js [hunk 1]
```diff
@@ -14,9 +14,17 @@ export default class Hyperlinking {
...
+                    <HyperlinkingTooltip position={selectionPosition} />,
+                    this.quill.container,
+                    this.quill.container.lastChild,
+                );
+
+                this.quill.format('highlight', true);
             } else {
                 render(null, this.quill.container, this.quill.container.lastChild);
+
+                this.quill.formatText(0, this.quill.getLength(), 'highlight', false);
```
Comment (latata) [inline src/modules/Hyperlinking.js]: smart!

File: src/blots/Highlight.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import Inline from 'quill/blots/inline';
+
+import './Highlight.scss';
+
+export default class Highlight extends Inline {
+    static blotName = 'highlight';
+
+    static className = 'pe-highlight';
+
+    static tagName = 'strong';
```
Comment (latata) [inline src/blots/Highlight.js]: can't we have just `span` here?

File: src/blots/Highlight.scss [hunk 3]
```diff
@@ -0,0 +1,7 @@
+$highlight-color: rgba(0, 183, 224, .3);
+
+.pe-highlight {
+	background-color: $highlight-color;
+	font-weight: normal;
+	outline: 2px solid $highlight-color;
```
Comment (latata) [inline src/blots/Highlight.scss:6]: why do we need outline?

File: src/modules/Hyperlinking.js (hunk 1 above)
Comment (latata) [reply src/modules/Hyperlinking.js]: I wonder what will happen when someone clicks submit button when text is selected. What will happen first remove highlighting or submit form?

**Review by latata — CHANGES_REQUESTED** [review]

File: src/modules/Hyperlinking.js (hunk 1 above)
Comment (xkxd) [reply src/modules/Hyperlinking.js]: don't know 🙈

File: src/blots/Highlight.js (hunk 2 above)
Comment (xkxd) [reply src/blots/Highlight.js]: span is broken, I've added a comment

File: src/blots/Highlight.scss (hunk 3 above)
Comment (xkxd) [reply src/blots/Highlight.scss:6]: by default background applies only for area that equals line height, based on designs I assumed we need something a bit wider and higher and `outline` comes in handy as it does not increase the actual size

Comment (latata) [reply src/blots/Highlight.scss:6]: cool cool cool

File: src/blots/Highlight.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+
+import './Highlight.scss';
+
+export default class Highlight extends Inline {
+    static blotName = 'highlight';
+
+    static className = 'pe-highlight';
+
+    // Span is broken, see https://github.com/quilljs/quill/issues/1866
+    static tagName = 'strong';
```
Comment (latata) [inline src/blots/Highlight.js:11]: what about div with `display: inline`?

**Review by latata — APPROVED** [review]

File: src/blots/Highlight.js (hunk 4 above)
Comment (xkxd) [reply src/blots/Highlight.js:11]: `strong` appears as more semantic in that case

**Review by latata — APPROVED** [review]

**Pull request merged**